src/dmd/mars.d

```diff
@@ -241,12 +241,12 @@ private int tryMain(size_t argc, const(char)** argv, ref Param params)
     bool is64bit = arch[0] == '6';
 
     version(Windows) // delete LIB entry in [Environment] (necessary for optlink) to allow inheriting environment for MS-COFF
-        if (is64bit || strcmp(arch, "32mscoff") == 0)
+        if (is64bit || arch == "32mscoff")
             environment.update("LIB", 3).value = null;
 
     // read from DFLAGS in [Environment{arch}] section
     char[80] envsection = void;
-    sprintf(envsection.ptr, "Environment%s", arch);
+    sprintf(envsection.ptr, "Environment%s", arch.ptr);
```

Comment:
Lgtm after a very superficial glance, except for this here - please use `%.*s` so as not to require undocumented null-termination.